rmi/servercode/TransImpl/TransactionManager.java

```diff
@@ -155,7 +167,9 @@ public void startDetector() {
 
 	public void enlist(int id, ResourceManager rm) throws RemoteException, TransactionAbortedException {
 		Transaction transaction = transactions.get(id);
-		transaction.add(rm);
+
+		// Add rm to transaction, with it's associated Vault hash key
```

Comment:
When we enlist an RM to the transaction, we generate a new hashKey for the temporary shadow of the RM.